waftools/checks/generic.py

```diff
@@ -186,3 +187,13 @@ def load_fragment(fragment):
     fragment_code = fp.read()
     fp.close()
     return fragment_code
+
+def check_macos_sdk(version):
+    def fn(ctx, dependency_identifier):
```

Comment:
Just a future note that in places like this early exit might be a worthwhile thing. Something a la:
```python
if not ctx.env.env.MACOS_SDK_VERSION or StrictVersion(ctx.env.MACOS_SDK_VERSION) < StrictVersion(version):
    return False

ctx.define(inflector.define_key(dependency_identifier), 1)
return True
```